src/conv/k-g/region_list.cpp

```diff
@@ -49,16 +49,58 @@ Bot& RegionList::addRegion
 }
 
 
+void RegionList::setAttributes
+(
+    const std::string&                        name,
+    const std::map<std::string, std::string>& attributes
+) {
+    m_list[name].attributes = attributes;
+}
+
+static void  writeAttributes
+(
+    rt_wdb* wdbp,
+    const char* name,
+    const std::map<std::string, std::string>& attributes
+) {
+
+    struct rt_db_internal     region_internal;
+
+    struct db_i*      dbip = wdbp->dbip;
+    struct directory* dp = db_lookup(dbip, name, 0);
+
+    if (dp == RT_DIR_NULL) {
+	bu_log("writeAttributes() Cannot find %s\n", name);
+	return;
+    }
+
+    if (rt_db_get_internal(&region_internal, dp, dbip, NULL, &rt_uniresource) <= 0) {

```

Comment:
The documentation says that <0 means error, i.e. 0 would be okay too. In addition, even in case of error, *region_internal* has to be freed. Therefore, why not
```c++
if (rt_db_get_internal() >= 0) {
    // write the attributes
    // update database
}
else
    // error message

// free region_internal
```

Reply:
I did see in the code that most of the checks where rt_db_get_internal()<0, I think I should read more the documentaion is it the hacking guid you are talking about?. you are right the check you did is better I will modify.

Reply:
I can't update the database immediatly db5_update_attributes() should be called after having called bu_avs_add() which requires having the rt_db_internal already checked.